app/code/Magento/InventoryCatalog/Observer/DeleteSourceItemsAfterProductDeletingObserver.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Observer;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\InventoryApi\Api\SourceItemRepositoryInterface;
+use Magento\Inventory\Model\SourceItem\Command\SourceItemsDelete as SourceItemsDeleteCommand;
+
+/**
+ * Delete source items related to product after product deleting
+ */
+class DeleteSourceItemsAfterProductDeletingObserver implements ObserverInterface
```

Comment:
Need to cover this behaviour with api-functional test
If you need help pls ping me in Slack